ui/component/or-attribute-input/src/or-anomaly-config-chart.ts

```diff
@@ -0,0 +1,303 @@
+                    return  ctx.p0.parsed.x > Date.now()? [2, 4] : undefined
+                })
+            },
+            fill: false,
+            data: [],
+        };
+        if(this.anomalyConfig){
+            response = await manager.rest.api.AnomalyDetectionResource.getAnomalyDatapointLimits(asset.id, attribute.name, this.anomalyConfig);
+            if (response.status === 200) {
+                if(response.data.length === 0){
+                    this.errorMessage = i18next.t("anomalyDetection.noDatapointsSaved");
+                    return datasets;
+                }else if(response.data.length === 1){
+                    this.errorMessage = i18next.t("anomalyDetection.notEnoughDatapointsSaved");
+                    return datasets;
+                }else if(response.data.length === 4 && response.data[1].findIndex(dp => dp != null)=== -1){
+                    this.errorMessage = i18next.t("anomalyDetection.notEnoughDatapointsSavedForLimits");
+                }
+                dataset.data = response.data[2].filter(value => value !== null && value !== undefined) as ScatterDataPoint[];
+                anomalyDataset.data = response.data[3].filter(value => value !== null) as ScatterDataPoint[];
+                let limitSectionCount = 0;
+                while(response.data[1].length > 1) {
+                    let minimaxdata = response.data[1].slice(response.data[1].findIndex(d => d !== null),response.data[1].length);
+                    let index = minimaxdata.findIndex(d => d === null);
+                    minimaxdata = minimaxdata.slice(0,index=== -1? minimaxdata.length:index);
+                    maxData.data = minimaxdata.filter(value => value !== null && value !== undefined) as ScatterDataPoint[];
+                    datasets.push(JSON.parse(JSON.stringify(maxData)));
+                    response.data[1] =  response.data[1].splice(response.data[1].findIndex(d => d !== null) + minimaxdata.length, response.data[1].length);
+                    limitSectionCount++
+                }
+                let i = 1
+                datasets.forEach(ds =>{
+                    ds.fill = limitSectionCount +i
+                    i++;
+                })
+                datasets.push(dataset);
+
+                while(response.data[0].length > 1){
+                    let minimindata = response.data[0].slice(response.data[0].findIndex(d => d !== null),response.data[0].length);
+                    let index = minimindata.findIndex(d => d === null);
+                    minimindata = minimindata.slice(0,index=== -1? minimindata.length:index);
+                    minData.data = minimindata.filter(value => value !== null && value !== undefined) as ScatterDataPoint[];
+                    datasets.push(JSON.parse(JSON.stringify(minData)));
+                    response.data[0] = response.data[0].splice(response.data[0].findIndex(d => d !== null) + minimindata.length, response.data[0].length);
```

Comment:
Can we merge these two `while` loops into one?

Reply:
probably but I can barley understand their functionality now. if merged into one readability will go down.